engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputUtils.java

```diff
@@ -0,0 +1,758 @@
+    }
+
+    return strings.toArray( new String[strings.size()] );
+  }
+
+  public static final String getLine( LogChannelInterface log, InputStreamReader reader, int formatNr,
+      StringBuilder line ) throws KettleFileException {
+    EncodingType type = EncodingType.guessEncodingType( reader.getEncoding() );
+    return getLine( log, reader, type, formatNr, line );
+  }
+
+  public static final String getLine( LogChannelInterface log, InputStreamReader reader, EncodingType encodingType,
+      int formatNr, StringBuilder line ) throws KettleFileException {
+    int c = 0;
+    line.setLength( 0 );
+    try {
+      switch ( formatNr ) {
+        case TextFileInputMeta.FILE_FORMAT_DOS:
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isReturn( c ) || encodingType.isLinefeed( c ) ) {
+              c = reader.read(); // skip \n and \r
+              if ( !encodingType.isReturn( c ) && !encodingType.isLinefeed( c ) ) {
+                // make sure its really a linefeed or cariage return
+                // raise an error this is not a DOS file
+                // so we have pulled a character from the next line
+                throw new KettleFileException( BaseMessages.getString( PKG, "TextFileInput.Log.SingleLineFound" ) );
+              }
+              return line.toString();
+            }
+            if ( c >= 0 ) {
+              line.append( (char) c );
+            }
+          }
+          break;
+        case TextFileInputMeta.FILE_FORMAT_UNIX:
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isLinefeed( c ) || encodingType.isReturn( c ) ) {
+              return line.toString();
+            }
+            if ( c >= 0 ) {
+              line.append( (char) c );
+            }
+          }
+          break;
+        case TextFileInputMeta.FILE_FORMAT_MIXED:
+          // in mixed mode we suppose the LF is the last char and CR is ignored
+          // not for MAC OS 9 but works for Mac OS X. Mac OS 9 can use UNIX-Format
+          while ( c >= 0 ) {
+            c = reader.read();
+
+            if ( encodingType.isLinefeed( c ) ) {
+              return line.toString();
+            } else if ( !encodingType.isReturn( c ) ) {
+              if ( c >= 0 ) {
+                line.append( (char) c );
+              }
+            }
+          }
+          break;
+        default:
+          break;
+      }
+    } catch ( KettleFileException e ) {
+      throw e;
+    } catch ( Exception e ) {
+      if ( line.length() == 0 ) {
+        throw new KettleFileException( BaseMessages.getString( PKG, "TextFileInput.Log.Error.ExceptionReadingLine", e
+            .toString() ), e );
+      }
+      return line.toString();
+    }
+    if ( line.length() > 0 ) {
+      return line.toString();
+    }
+
+    return null;
+  }
+
+  public static final Object[] convertLineToRow( LogChannelInterface log, TextFileLine NewFileLine,
```

Comment:
A couple of argument variables here have bad names:
s/NewFileLine/textFileLine
s/additionalOutputFIelds/additionalOutputFields